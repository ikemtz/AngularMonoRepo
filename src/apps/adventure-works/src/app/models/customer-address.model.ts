/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IAddress } from './address.model';
import { ICustomer } from './customer.model';

export interface ICustomerAddress {
  id?: string;
  customerId?: string;
  addressId?: string;
  addressType?: string;
  address?: IAddress;
  customer?: ICustomer;
}
