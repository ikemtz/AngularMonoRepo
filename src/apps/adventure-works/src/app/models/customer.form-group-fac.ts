/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function CustomerFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    nameStyle: new FormControl('', Validators.required),
    title: new FormControl('', Validators.maxLength(8)),
    firstName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(50)])),
    middleName: new FormControl('', Validators.maxLength(50)),
    lastName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(50)])),
    suffix: new FormControl('', Validators.maxLength(10)),
    companyName: new FormControl('', Validators.maxLength(128)),
    salesPerson: new FormControl('', Validators.maxLength(256)),
    emailAddress: new FormControl('', Validators.maxLength(50)),
    phone: new FormControl('', Validators.maxLength(50)),
    customerAddresses: new FormArray([]),
    saleOrders: new FormArray([]),
  });
};
