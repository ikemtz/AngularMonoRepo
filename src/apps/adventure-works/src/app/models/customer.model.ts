/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { ICustomerAddress } from './customer-address.model';
import { ISaleOrder } from './sale-order.model';

export interface ICustomer {
  id?: string;
  nameStyle?: boolean;
  title?: string;
  firstName?: string;
  middleName?: string;
  lastName?: string;
  suffix?: string;
  companyName?: string;
  salesPerson?: string;
  emailAddress?: string;
  phone?: string;
  customerAddresses?: ICustomerAddress[];
  saleOrders?: ISaleOrder[];
}
