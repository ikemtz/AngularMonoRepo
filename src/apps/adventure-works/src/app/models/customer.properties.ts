/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum CustomerProperties {
  ID = 'id',
  NAME_STYLE = 'nameStyle',
  TITLE = 'title',
  FIRST_NAME = 'firstName',
  MIDDLE_NAME = 'middleName',
  LAST_NAME = 'lastName',
  SUFFIX = 'suffix',
  COMPANY_NAME = 'companyName',
  SALES_PERSON = 'salesPerson',
  EMAIL_ADDRESS = 'emailAddress',
  PHONE = 'phone',
  CUSTOMER_ADDRESSES = 'customerAddresses',
  SALE_ORDERS = 'saleOrders',
}
