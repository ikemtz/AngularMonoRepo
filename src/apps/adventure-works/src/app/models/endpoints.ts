/**
 * This file is generated by the openapi-ts-generator
 * #endpoints-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum Endpoints {
  addresses = 'aw-odata/odata/v1/Addresses',
  customers = 'aw-odata/odata/v1/Customers',
  product_categories = 'aw-odata/odata/v1/ProductCategories',
  product_models = 'aw-odata/odata/v1/ProductModels',
  products = 'aw-odata/odata/v1/Products',
  sale_order_details = 'aw-odata/odata/v1/SaleOrderDetails',
  sale_orders = 'aw-odata/odata/v1/SaleOrders',
}
