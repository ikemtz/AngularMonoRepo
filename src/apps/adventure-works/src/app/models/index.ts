/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './address.form-group-fac';
export * from './address.model';
export * from './address.properties';
export * from './customer-address.form-group-fac';
export * from './customer-address.model';
export * from './customer-address.properties';
export * from './customer.form-group-fac';
export * from './customer.model';
export * from './customer.properties';
export * from './endpoints';
export * from './product-category.form-group-fac';
export * from './product-category.model';
export * from './product-category.properties';
export * from './product-model.form-group-fac';
export * from './product-model.model';
export * from './product-model.properties';
export * from './product.form-group-fac';
export * from './product.model';
export * from './product.properties';
export * from './sale-order-detail.form-group-fac';
export * from './sale-order-detail.model';
export * from './sale-order-detail.properties';
export * from './sale-order.form-group-fac';
export * from './sale-order.model';
export * from './sale-order.properties';
