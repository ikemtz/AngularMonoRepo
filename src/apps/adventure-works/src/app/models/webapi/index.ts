/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './customer-address.form-group-fac';
export * from './customer-address.model';
export * from './customer-address.properties';
export * from './customer.form-group-fac';
export * from './customer.model';
export * from './customer.properties';
export * from './order-address.form-group-fac';
export * from './order-address.model';
export * from './order-address.properties';
export * from './order-line-item.form-group-fac';
export * from './order-line-item.model';
export * from './order-line-item.properties';
export * from './order-status-types.enum';
export * from './order.form-group-fac';
export * from './order.model';
export * from './order.properties';
export * from './product-category.form-group-fac';
export * from './product-category.model';
export * from './product-category.properties';
export * from './product-model.form-group-fac';
export * from './product-model.model';
export * from './product-model.properties';
export * from './product.form-group-fac';
export * from './product.model';
export * from './product.properties';
export * from './sales-agent.form-group-fac';
export * from './sales-agent.model';
export * from './sales-agent.properties';
export * from './shipping-types.enum';
