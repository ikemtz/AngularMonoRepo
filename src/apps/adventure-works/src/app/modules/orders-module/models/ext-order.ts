import { PagerSettings } from '@progress/kendo-angular-grid';
import { ODataResult, ODataState } from 'imng-kendo-odata';
import { IOrder, IOrderLineItem } from '../../../models/odata';

export interface IExtOrder extends IOrder {
  orderLineItemODataState: ODataState;
  orderLineItemOData: ODataResult<IOrderLineItem>;
  orderLineItemPagerSettings: PagerSettings | false;
}
