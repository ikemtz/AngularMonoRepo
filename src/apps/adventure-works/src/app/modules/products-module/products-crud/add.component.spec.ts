import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';

import { ProductAddComponent } from './add.component';
import { ProductCrudFacade } from './crud.facade';
import { IProduct, ProductProperties } from '../../../models/webapi';
import { of } from 'rxjs';
import { readFirst } from 'imng-ngrx-utils/testing';
import { DatePickerModule } from '@progress/kendo-angular-dateinputs';
import { DropDownsModule } from '@progress/kendo-angular-dropdowns';

export function createMockProductFacade() {
  return {
    currentEntity$: of({}),
    productModels$: of([
      { id: 'abc', name: 'abc', description: 'abc', },
      { id: 'xyz', name: 'xyz', description: 'xyz', },]),
    loadProductModels: jest.fn(),
    productCategories$: of([
      { id: 'abc', name: 'abc', },
      { id: 'xyz', name: 'xyz', },]),
    loadProductCategories: jest.fn(),
  };
}
describe('ProductAddComponent', () => {
  let component: ProductAddComponent;
  let fixture: ComponentFixture<ProductAddComponent>;
  let facade: ProductCrudFacade;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [ProductAddComponent],
      imports: [ReactiveFormsModule, NoopAnimationsModule, DatePickerModule, DropDownsModule],
      providers: [{
        provide: ProductCrudFacade, useValue: createDataEntryMockFacade(createMockProductFacade())
      }],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ProductAddComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(ProductCrudFacade);
    fixture.detectChanges();
  });

  afterAll(() => {
    component.ngOnDestroy();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should save', () => {
    component.initForm();
    component.addEditForm?.patchValue({
      [ProductProperties.ID]: 'ID',
      [ProductProperties.NAME]: 'NAME',
      [ProductProperties.NUM]: 'NUM',
      [ProductProperties.COLOR]: 'COLOR',
      [ProductProperties.STANDARD_COST]: 0,
      [ProductProperties.LIST_PRICE]: 0,
      [ProductProperties.SIZE]: 'SIZE',
      [ProductProperties.WEIGHT]: 0,
      [ProductProperties.PRODUCT_CATEGORY_ID]: 'PRODUCT_CATEGORY_ID',
      [ProductProperties.PRODUCT_MODEL_ID]: 'PRODUCT_MODEL_ID',
      [ProductProperties.SELL_START_DATE]: new Date(),
      [ProductProperties.SELL_END_DATE]: new Date(),
      [ProductProperties.DISCONTINUED_DATE]: new Date(),
      [ProductProperties.THUMB_NAIL_PHOTO]: 'THUMB_NAIL_PHOTO',
      [ProductProperties.PRODUCT_MODEL]: 'PRODUCT_MODEL',
      [ProductProperties.PRODUCT_CATEGORY]: 'PRODUCT_CATEGORY',
    });

    let item: IProduct | undefined;
    facade.saveNewEntity = jest.fn(x => (item = x));
    facade.updateExistingEntity = jest.fn();
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(1);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);

    expect(item).toMatchSnapshot({
      sellStartDate: expect.any(Date),
      sellEndDate: expect.any(Date),
      discontinuedDate: expect.any(Date),
    });

  });

  test('should not save', () => {
    component.addEditForm = { valid: false } as never;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);
  });

  test('should cancel', () => {
    facade.clearCurrentEntity = jest.fn();
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });

  test('should support ProductModel filters', async () => {
    component.handleProductModelFilter('xy');
    const result = await readFirst(component.productModels$);
    expect(result).toStrictEqual([{ id: 'xyz', name: 'xyz', description: 'xyz', }]);
  });

  test('should support ProductCategory filters', async () => {
    component.handleProductCategoryFilter('xy');
    const result = await readFirst(component.productCategories$);
    expect(result).toStrictEqual([{ id: 'xyz', name: 'xyz', }]);
  });
});
