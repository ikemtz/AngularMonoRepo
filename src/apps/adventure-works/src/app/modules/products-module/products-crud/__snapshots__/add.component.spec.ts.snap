// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAddComponent should save 1`] = `
Object {
  "color": "COLOR",
  "discontinuedDate": Any<Date>,
  "id": undefined,
  "listPrice": 0,
  "name": "NAME",
  "num": "NUM",
  "orderLineItems": Array [],
  "productCategory": "PRODUCT_CATEGORY",
  "productCategoryId": "PRODUCT_CATEGORY_ID",
  "productModel": "PRODUCT_MODEL",
  "productModelId": "PRODUCT_MODEL_ID",
  "sellEndDate": Any<Date>,
  "sellStartDate": Any<Date>,
  "size": "SIZE",
  "standardCost": 0,
  "thumbNailPhoto": "THUMB_NAIL_PHOTO",
  "weight": 0,
}
`;
